src/nni_manager/training_service/reusable/environments/amlEnvironmentService.ts

```diff
@@ -98,8 +98,7 @@ export class AMLEnvironmentService extends EnvironmentService {
                     environment.setFinalStatus('SUCCEEDED');
                     break;
                 case 'FAILED':
-                    environment.setFinalStatus('FAILED');
-                    break;
+                    return Promise.reject(`AML: job ${environment.jobId} is failed!`);
```

Comment:
besides returning reject, do we need to set final status here?

Reply:
don't need to set status for environment here. when return Promise.reject, the experiment become `error` status, and nniManager stops. this final status is not useful anymore.

Reply:
got it

Reply:
well, the status is not visible so far. But if we expose it to webui in future, it would be a hole.

Reply:
agree

Reply:
updated.